test.sh

```diff
@@ -500,6 +500,11 @@ function test_docs {
   diff docs/reference/includes $kb_orig/test/docs/expected/includes
   diff docs/reference/manifest.json $kb_orig/test/docs/expected/manifest.json
   diff docs/reference/config.yaml $kb_orig/test/docs/expected/config.yaml
+
+  header_text "testing doc annotations"
+  sed -i -e "s|// +kubebuilder:categories=foo,bar|// +kubebuilder:categories=foo,bar\n// +kubebuilder:doc:note=test notes message annotations\n// +kubebuilder:doc:warning=test warnings message annotations|" pkg/apis/insect/v1beta1/bee_types.go
```

Comment:
why not updating the file directly. running the script will leave the tree dirty. I think `sed`ding it is very hacky.

Reply:
By far, the scripts is only used for dev purpose. A series of tests are going to be added along with new features. Will consider a better way for e2e test.

Reply:
@anfernee agreed this isn't is a total hack, but something we have relied upon while we setup better better infrastructure for testing.

We are working on the integration / e2e test infrastructure this Sprint, and will migrate these tests.